snuba/query/mql/parser.py

```diff
@@ -1116,6 +1109,17 @@ def parse_mql_query(
             settings,
         )
 
+    # Filter in select optimizer
+    feat_flag = get_int_config("enable_filter_in_select_optimizer", default=1)
+    if feat_flag:
+        with sentry_sdk.start_span(
+            op="processor", description="filter_in_select_optimize"
+        ):
+            if settings is None:
+                FilterInSelectOptimizer().process_query(query, HTTPQuerySettings())
```

Comment:
I would like to point out that in this context settings can be None, LogicalQueryProcessor interface requires settings not None, thus I had to create dummy object.

Since settings arent used by my processor it doesnt matter, but the fact that i have to create this dummy instead of None smells to me